components/automate-ui/src/app/helpers/auth/id-mapper.ts

```diff
@@ -1,6 +1,6 @@
 export class IdMapper {
 
   public static transform(input: string): string {
-    return input.toLowerCase().replace(/[^a-z0-9]/g, '-');
+    return input.toLowerCase().replace(/[^a-z0-9_]/g, '-');
```

Comment:
This IdMapper is used by `CreateObjectModalComponent` to generate the ID corresponding to the name as it is typed by the user.
See: https://github.com/chef/automate/blob/16316b40e8d9103a3157b66b6e96301c9a0b1780/components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts#L68-L74